flixel/system/FlxSplash.hx

```diff
@@ -199,7 +199,7 @@ class FlxSplash extends FlxState
 		#end
 		FlxG.stage.removeChild(_sprite);
 		FlxG.stage.removeChild(_text);
-		FlxG.switchState(Type.createInstance(nextState, []));
+		FlxG.switchState(() -> Type.createInstance(nextState, []));
```

Comment:
nextState should also be a `()->FlxState`, and the FlxGame constructor should take a function instead of a Class Ref

Reply:
Hm So like making the PlayState.new also in FlxGame?
I liked that you didn't need to add anything, just the name but ok

Reply:
where before you had
```hx
addChild(new FlxGame(400, 300, MenuState);
```
now you can just do
```hx
addChild(new FlxGame(400, 300, MenuState.new);
```

the reason for this is to allow constuctor args, like
```hx
addChild(new FlxGame(400, 300, MenuState.new.bind(arg);
```